File: Cenfomon/Capa_Logica/src/main/java/com/cenfomon/Cenfomon_Capa_Logica/Potion.java
package com.cenfomon.Cenfomon_Capa_Logica;

public class Potion extends Items {

    private String strenght;
    private int idCenfomon;

    public Potion(int id, String location, float cost, String name, String imgURL, String strenght, int idCenfomon) {
        super(id, location, cost, name, imgURL);
        this.strenght = strenght;
        this.idCenfomon = idCenfomon;
    }

    public Potion(){}

    public String getStrenght() {
        return strenght;
    }

    public void setStrenght(String strenght) {
        this.strenght = strenght;
    }

    public int getIdCenfomon() {
        return idCenfomon;
    }

    public void setIdCenfomon(int idCenfomon) {
        this.idCenfomon = idCenfomon;
    }
}

File: Cenfomon/Cenfomon_COI/bl/factoryMethodItems/productoConcreto/Potion.ts
import {Item} from "../Producto/Item";

export class Potion extends Item{

    constructor(_id: number, _location: Location, _cost: number, _name: string, _imgURL: string, private _strength: string, private _idCenfomon: number) {
        super(_id, _location, 90, _name, _imgURL);
        this._idCenfomon = _idCenfomon;
        this._strength = _strength;
    }

    generarID(): {} {
        return {};
    }

    
    get strength(): string {
        return this._strength;
    }

    set strength(value: string) {
        this._strength = value;
    }

    get idCenfomon(): number {
        return this._idCenfomon;
    }

    set idCenfomon(value: number) {
        this._idCenfomon = value;
    }
}
